src/LuaAST.ts

```diff
@@ -139,13 +141,19 @@ export function setParent(node: Node | Node[] | undefined, parent: Node): void
                 n.pos = parent.pos;
                 n.end = parent.end;
             }
+            if (!n.tsOriginal) {
+                n.tsOriginal = parent.tsOriginal;
+            }
         });
     } else {
         node.parent = parent;
         if (node.pos === -1 || node.end === -1) {
             node.pos = parent.pos;
             node.end = parent.end;
         }
+        if (!node.tsOriginal) {
+            node.tsOriginal = parent.tsOriginal;
+        }
```

Comment:
It is intentionally avoided to link the Lua AST node to the tsOriginal. Is this really required?

Reply:
I think I can rework things to avoid it, but I'm curious what the reason is for avoiding it? Seems like it would be quite useful.